teuthology/suite/__init__.py

```diff
@@ -86,11 +88,55 @@ def expand_short_repo_name(name, orig):
     # otherwise, assume a full URL
     return name
 
+def calc(args):
+    incl_keys=[]
+    excl_keys=[]
+    if args['--filter']:
+        incl_keys = args['--filter'].split(',')
+        log.debug("Using include filter: {}".format(incl_keys))
+    if args['--filter-out']:
+        excl_keys = args['--filter-out'].split(',')
+        log.debug("Using exclude filter: {}".format(excl_keys))
+
+    _seed=int(args['--seed'])
```

Comment:
Shouldn't we be using some default value in case the seed just wasn't set? Setting it in the "normal" path is definitely weird but also a little more complicated than we're doing here...

Reply:
@gregsfortytwo the default value for seed is -1, if the default value is given there is code which generates a random seed value, which is not what we want sine the seed will be different each time. So I just set to None default value.

Reply:
However, I tried to use run --calc with different --seed value and it looks like nothing is changed, though it does not generate different Cycle sequence with combination --subset, it is not critical right now.

Reply:
Do we actually go through that default path when running ```--calc```?
Granted I don't really understand how the seed is used here but it looks like it's only interpreted in the rerun path, so since we're shortcutting everything except for ```process_args``` we can't make any assumptions what it looks like.